api/v1beta1/frrconfiguration_types.go

```diff
@@ -95,9 +95,20 @@ type Import struct {
 // Neighbor represents a BGP Neighbor we want FRR to connect to.
 type Neighbor struct {
 	// ASN is the AS number to use for the local end of the session.
+	// ASN and DynamicASN are mutually exclusive and one of them must be specified.
 	// +kubebuilder:validation:Minimum=0
 	// +kubebuilder:validation:Maximum=4294967295
-	ASN uint32 `json:"asn"`
+	// +optional
+	ASN uint32 `json:"asn,omitempty"`
+
+	// DynamicASN detects the AS number to use for the local end of the session
+	// without explicitly setting it via the ASN field. Limited to:
+	// internal - if the neighbor's ASN is different than the router's the connection is denied.
+	// external - if the neighbor's ASN is the same as the router's the connection is denied.
+	// ASN and DynamicASN are mutually exclusive and one of them must be specified.
+	// +kubebuilder:validation:Enum=internal;external
```

Comment:
Can we enforce the mutual exclusivity (or -ness) with CEL rules?
https://kubernetes.io/docs/reference/using-api/cel/

Reply:
tried for a while, did not go so well 😅  I guess we can tackle it in a different time (applies for password and passwordSecret as well)